### PR TITLE
Refactor coffeegram validation

### PR DESCRIPTION
**Pull request by manuscriptmastr**
@nybblr

### CODE REVIEW

File: src/routes.js [hunk 1]
```diff
@@ -35,17 +36,30 @@ router.get('/coffeegrams/new', async ctx => {
...
+  }
+
   await Coffeegram.create({
     userId: ctx.state.currentUser.id,
-    image: path.basename(form.image[0].path),
-    description: form['description'],
-    type: form['type'],
-    shop: form['shop']
+    image: path.basename(image[0].path),
+    description,
```
Comment (nybblr) [inline src/routes.js]: Uh oh, these won't be cleaned anymore and you've opened yourself to security issues. I think you want to do `description: coffeeParams.description`?

File: src/validation.js [hunk 2]
```diff
@@ -10,4 +11,17 @@ const validate = ({name, email, username, passwordFirst, passwordConfirmation})
   return errors;
 }
 
-module.exports = validate;
+const validateImage = ({ image }) => {
+  var errors = {};
+  if (isEmpty(image[0].size)) {
```
Comment (nybblr) [inline src/validation.js]: `isEmpty` is the wrong method, just test if size === 0.

Comment (manuscriptmastr) [reply src/validation.js]: Done.

**Comment by nybblr** [comment]
:+1:

**Comment by nybblr** [comment]
:+1:

**Pull request merged**